support/signal.c

```diff
@@ -481,13 +481,6 @@ wait_for_any (signal_info** signals, int count, int *currfd, struct pollfd* fd_s
 	// Poll until one of this signal_info's pipes is ready to read.
 	// Once a second, stop to check if the VM is shutting down.
 	do {
-		struct timeval tv;
-		struct timeval *ptv = NULL;
-		if (timeout != -1) {
-			tv.tv_sec  = timeout / 1000;
-			tv.tv_usec = (timeout % 1000)*1000;
-			ptv = &tv;
-		}
 		r = poll (fd_structs, count, timeout);
```

Comment:
Is `poll` expecting `ptv`?

Reply:
Er, nevermind that, But the intent was to loop while `time remaining` and reduce it as it goes?

Reply:
With select() it was called with same timeout every time, same as it is now with poll(). So I'm not sure what you're suggesting.